### PR TITLE
Add CreateDatabaseTarget and DeleteDatabaseTarget to TargetsService

### PR DESCRIPTION
**Pull request by roshbhatia**
Adds `TargetsService.CreateDatabaseTarget`, `TargetsService.DeleteDatabaseTarget` and associated structs, trying to follow the pattern in [the EnvironmentsService](https://github.com/bastionzero/bastionzero-sdk-go/blob/b0a6992ab290a631c360010a9ac22e932cf5834b/bastionzero/service/environments/environments.go#L25).

Looks like there are no unit tests in this project, but to validate functionality I used https://gist.github.com/roshbhatia/570518a07ccfbb8b131ab085364eba3b.

### CODE REVIEW

**Comment by roshbhatia** [comment]
@ymarcus93 Do I need to do anything to kick off the platform tests?

File: bastionzero/service/targets/database.go
Comment (ymarcus93) [inline bastionzero/service/targets/database.go:29]: For struct types that are only used in the _create_ endpoint methods, I'm trying to only include `omitempty` if the field is optional as defined by the API spec. Additionally, I make the field a pointer type if the default/zero value has semantic meaning that is different than null/unspecified.

So with that in mind, can you make the following changes?

- Remove `omitempty` from `TargetName`, `ProxyTargetID`, `RemoteHost`, and `RemotePort` since these fields are required
- Remove the pointer type for `RemotePort`: `RemotePort *Port` --> `RemotePort Port`

**Review by ymarcus93 — COMMENTED** [review]
Thanks for the contribution! I've left some comments and once resolved we can merge this and have it published in the next version. Next week, I'll be working on adding all the other remaining API endpoints, but thanks for making it a bit easier for me! Also, regarding the lack of unit tests, this is in our backlog to add some to this project.

Also, the CI workflow can only run after approved by a maintainer; I'll run it once the other changes are pushed.

File: bastionzero/service/targets/database.go
Comment (roshbhatia) [reply bastionzero/service/targets/database.go:29]: @ymarcus93, should I remove the pointer type from `LocalPort` as well? Looking at https://cloud.bastionzero.com/api/#get-/api/v2/targets/database/-id-, doesn't seem like the zero value has a semantic meaning.

File: bastionzero/service/targets/database.go
Comment (ymarcus93) [inline bastionzero/service/targets/database.go:33]: Typo in comment: `CreateDatabaseTargetRequest` --> `CreateDatabaseTargetResponse`

**Review by ymarcus93 — COMMENTED** [review]
Looks good. Just a small typo and then we're good to merge

**Review by ymarcus93 — APPROVED** [review]

**Pull request merged**
